User interface rework of settings and "About" dialog

phoerious opened the pull request:
## Description
<!--- Describe your changes in detail -->
After @louib provided a good starting point in #288 (kudos!), I decided to finally start taking on my planned user interface rework of the following components:

- the ugly vertical tab bar which is used by the entry/group edit widgets and the settings
- the confusing and unordered application settings (reorder and layout widgets, choose better default values and shorten some HTTP settings by moving explanations into tooltips)
- the very basic about dialog

I think this PR doesn't need any more words, so here are the screenshots:

### Settings on KDE
![settings1-kde](https://cloud.githubusercontent.com/assets/911270/23238580/efc5dac6-f962-11e6-94f2-f9eed014aec1.png)

![settings2-kde](https://cloud.githubusercontent.com/assets/911270/23238582/f1cd14ba-f962-11e6-8cc2-63b90ba14802.png)

![settings3-kde](https://cloud.githubusercontent.com/assets/911270/23238585/f45d0aa0-f962-11e6-8232-0d0dbc349dbd.png)

![settings4-kde](https://cloud.githubusercontent.com/assets/911270/23238587/f6c9ec18-f962-11e6-945f-828430f9a364.png)

![settings5-kde](https://cloud.githubusercontent.com/assets/911270/23238589/f8e34210-f962-11e6-85b9-44acbc32149e.png)

### Settings on OS X
![settings-osx](https://cloud.githubusercontent.com/assets/911270/23238590/fb051e56-f962-11e6-9a1e-d41f090e78a3.png)

### Settings on Windows
![settings-windows](https://cloud.githubusercontent.com/assets/911270/23238593/ff4e4d20-f962-11e6-8961-1e8fe8a55554.png)

### Edit Entry
![edit-entry-kde](https://cloud.githubusercontent.com/assets/911270/23238602/07263120-f963-11e6-9f9e-43a11aa62270.png)

### ...when there is not enough vertical space
![edit-entry-scroll-kde](https://cloud.githubusercontent.com/assets/911270/23238605/096c1d96-f963-11e6-8e4e-4c7f081ee37d.png)

### Edit Group
![edit-group-kde](https://cloud.githubusercontent.com/assets/911270/23240364/648fabb6-f96d-11e6-9314-dee64a9fdad2.png)

### About dialog
![about-dialog1](https://cloud.githubusercontent.com/assets/911270/23240192/47ff6816-f96c-11e6-876d-0c4be226091f.png)

![about-dialog2](https://cloud.githubusercontent.com/assets/911270/23239161/df1d60c4-f965-11e6-9147-63ad018c69d2.png)


## How Has This Been Tested?
Tested on KDE Plasma 5, Windows 10 and OS X 10.12. Not tested on Ubuntu.

## Types of changes
<!--- What types of changes does your code introduce? -->
<!--- Please remove all lines which don't apply. -->
- ✅ New feature (non-breaking change which adds functionality)

## Checklist:
<!--- Please go over all the following points. -->
<!--- Again, remove any lines which don't apply. -->
<!--- Pull Requests that don't fulfill all [REQUIRED] requisites are likely -->
<!--- to be sent back to you for correction or will be rejected.  -->
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

droidmonkey commented:
I will do a thorough review tomorrow. This looks amazing!

RealOrangeOne commented:
Looks great! 1 observation, in the settings screenshot, the left hand nav tabs are underlined, but dont seem to be anywhere else? Is this intentional? If so maybe all of _Browser Integration_ should be highlighted rather just _integration_?

phoerious commented:
That's part of the system style. Only the currently selected entry is underlined.

RealOrangeOne commented:
@phoerious I thought so. But should the other screens (eg edit entry) not have underlines too?

phoerious commented:
No. It's only underlined when it's selected and focused. On the other screens, the input fields are focused, not the category item.

louib commented:
@phoerious This is really nice! Some comments:

For the `Enable KeePassXC HTTP procotol` checkbox, do you think we could align the label with the textbox, and make a tooltip out of the help text (This is required...)?

Is the `Report bugs` url actually pointing to https://github.com/keepassxreboot/keepassxc/issues?

Could we reduce the width of the HTTP Port TextField? It seems really wide considering that the maximum length of the input is 5 characters. Not sure what would be the best width though...

Did you manage to fix the resizing bug in the settings on MacOS? Is it what `Fix funny resizing` fixes?

Again, good job!

louib commented on src/core/Config.cpp:
```diff
@@ -102,10 +102,11 @@ void Config::init(const QString& fileName)
     m_defaults.insert("security/clearclipboardtimeout", 10);
     m_defaults.insert("security/lockdatabaseidle", false);
-    m_defaults.insert("security/lockdatabaseidlesec", 10);
+    m_defaults.insert("security/lockdatabaseidlesec", 240);
```
@phoerious Is there a particular reason behind changing the default locking delay?

phoerious replied:
Yes. 10 seconds is bullshit. it's a good timeout for clearing the clipboard, but locking the after 10 seconds is utter garbage. If you want instant lock, enable "Lock when minimizing" and then just send the window to the background once you're done with it. In any other case you probably need some time for whatever you're doing. And if you're logging into all your services in the morning, you really don't want to enter your master key for each and every one of them, because the DB locked itself in the meantime.

louib commented on src/http/OptionDialog.cpp:
```diff
@@ -33,65 +39,48 @@ OptionDialog::~OptionDialog()
-    ui->checkBoxSpecialChars->setChecked(settings.passwordUseSpecial());
-    ui->checkBoxEnsureEvery->setChecked(settings.passwordEveryGroup());
-    ui->checkBoxExcludeAlike->setChecked(settings.passwordExcludeAlike());
-    ui->spinBoxLength->setValue(settings.passwordLength());
```
:+1:

louib commented on src/http/OptionDialog.cpp:
```diff
@@ -33,65 +39,48 @@ OptionDialog::~OptionDialog()
-    settings.setPasswordEveryGroup(ui->checkBoxEnsureEvery->isChecked());
-    settings.setPasswordExcludeAlike(ui->checkBoxExcludeAlike->isChecked());
-    settings.setPasswordLength(ui->spinBoxLength->value());
-*/
```
:+1:

phoerious commented:
> For the Enable KeePassXC HTTP procotol checkbox, do you think we could align the label with the textbox, and make a tooltip out of the help text (This is required...)?

I added a commit for that. I also chose to disable the HTTP settings tab widget if this box isn't checked.

> Is the Report bugs url actually pointing to https://github.com/keepassxreboot/keepassxc/issues?
Yes, of course.

> Could we reduce the width of the HTTP Port TextField? It seems really wide considering that the maximum length of the input is 5 characters. Not sure what would be the best width though...

I'm not sure what looks better. I didn't see it as an issue, but now that you say it... Really not sure which of both options is better. I added a commit which makes its horizontal policy "Preferred" for now. That causes the field to have the minimum width required to fit its placeholder ("Default port: 19455").

> Did you manage to fix the resizing bug in the settings on MacOS? Is it what Fix funny resizing fixes?

What resizing bug? The answer is probably yes, but I'm not sure which bug you mean.

droidmonkey commented:
I made a slight upgrade to the autotype UI. Otherwise this looks and feels awesome. Very well done.

![keepassxc_autotype_v2](https://cloud.githubusercontent.com/assets/2809491/23322551/7f6fb552-fab2-11e6-9e4f-dad4fdfc0fcd.png)

phoerious commented:
Nice little change.

phoerious commented:
Put some finishing touches on the about dialog. It now opens as a dialog (therefore you can't open it more than once anymore), doesn't appear so huge on Windows anymore (Windows has tiny font sizes) and on Mac it's a proper sheet now.

t4777sd commented:
I just wanted to say it looks awesome. Great work!
